pkg/config/proxyrule/rule.go

```diff
@@ -40,14 +40,80 @@ const (
 // requests to an optional set of checks, an optional set of updares, and an
 // optional filter.
 type Spec struct {
-	Locking    LockMode           `json:"lock,omitempty"`
-	Matches    []Match            `json:"match"`
-	Checks     []StringOrTemplate `json:"check,omitempty"`
-	Must       []StringOrTemplate `json:"must,omitempty"`
-	MustNot    []StringOrTemplate `json:"mustNot,omitempty"`
-	Updates    []StringOrTemplate `json:"update,omitempty"`
-	PreFilters []PreFilter        `json:"prefilter,omitempty"`
-	// TODO: PostFilter
+	// Locking is the locking mode for this rule. The default is specified on
+	// the command line as a flag.
+	//
+	// If set to "Optimistic", the proxy will use optimistic locking by attempting
+	// to perform the update and rolling back if the update fails.
+	//
+	// If set to "Pessimistic", the proxy will use pessimistic locking by
+	// acquiring a lock on the object before performing the update.
+	Locking LockMode `json:"lock,omitempty"`
+
+	// Matches defines the requests that this rule applies to. Cannot be empty.
+	Matches []Match `json:"match"`
+
+	// Checks are the authorization checks to perform, in SpiceDB, if the request matches.
+	// If empty, the request will be allowed without any checks.
+	Checks []StringOrTemplate `json:"check,omitempty"`
+
+	// PreFilters are LookupResources requests to filter the results before any
+	// authorization checks are performed. Used for List and Watch requests.
+	PreFilters []PreFilter `json:"prefilter,omitempty"`
+
+	// Update contains the updates to perform if the request matches, the checks succeed,
+	// and this is a write operation of some kind (Create, Update, or Delete).
+	Update Update `json:"update,omitempty"`
```

Comment:
Personally I find "write" to be the more generic word. Create, update, delete are all writes.